src/app/core/services/app-server/app-server.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from '@angular/core'
+import * as moment from 'moment-timezone'
+import * as Swagger from 'swagger-client'
+
+import {
+  DefaultAppServerURL,
+  DefaultRequestJSONContentType
+} from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { StorageKeys } from '../../../shared/enums/storage'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { SubjectConfigService } from '../config/subject-config.service'
+import { LocalizationService } from '../misc/localization.service'
+import { LogService } from '../misc/log.service'
+import { StorageService } from '../storage/storage.service'
+import { TokenService } from '../token/token.service'
+
+@Injectable()
+export class AppServerService {
+  private API_DOCS_URL = '/v3/api-docs'
+
+  RADAR_USER_CONTROLLER = 'radar-user-controller'
+  RADAR_PROJECT_CONTROLLER = 'radar-project-controller'
+  MAX_API_RETRIES = 10
+  apiClient: any
+
+  constructor(
+    public storage: StorageService,
+    public subjectConfig: SubjectConfigService,
+    public logger: LogService,
+    public remoteConfig: RemoteConfigService,
+    public localization: LocalizationService,
+    private token: TokenService
+  ) {}
+
+  init() {
+    return (this.apiClient
+      ? Promise.resolve()
+      : this.initApiClient()
+    ).then(() => this.checkProjectAndSubjectExistElseCreate())
+  }
+
+  initApiClient() {
+    return Promise.all([this.getAppServerURL(), this.token.getTokens()]).then(
+      ([url, tokens]) =>
+        Swagger({
+          url: url + this.API_DOCS_URL,
+          requestInterceptor: req => {
+            req.headers['Authorization'] = 'Bearer ' + tokens.access_token
+            req.headers['Content-Type'] = DefaultRequestJSONContentType
+          }
+        })
+          .then(client => (this.apiClient = client))
+          .catch(e => this.logger.error('Error pulling API docs', e))
+    )
+  }
+
+  getApiClient() {
+    if (this.apiClient) return Promise.resolve(this.apiClient)
+    else return this.initApiClient().then(() => this.apiClient)
+  }
+
+  checkProjectAndSubjectExistElseCreate(): Promise<any> {
+    return this.checkProjectExistsElseCreate().then(() =>
+      this.checkSubjectExistsElseCreate()
+    )
+  }
+
+  checkProjectExistsElseCreate(): Promise<any> {
+    // NOTE: Adding retries here because of random 'Failed to load resource'
```

Comment:
Sorry I also don't remember what was going on here.  😅 But this is probably just because of my local app server instance. I'll try to investigate this again.